src/modules/pmc/src/lib/PmcBase.cpp

```diff
@@ -561,3 +607,130 @@ std::string PmcBase::Trim(const std::string& str, const std::string& trim)
+{
+    std::string hashString = "";
+    std::string command = std::regex_replace(g_commandGetSourcesFingerprint, std::regex("\\$value"), sourcesDirectory);
+    int status = RunCommand(command.c_str(), &hashString);
+
+    if (status != MMI_OK && IsFullLoggingEnabled())
+    {
+        OsConfigLogError(PmcLog::Get(), "Get the fingerprint of source files in directory %s failed with status %d", sourcesDirectory, status);
+    }
+
+    return !hashString.empty() ? hashString : "(failed)";
+}
+
+std::vector<std::string> PmcBase::ListFiles(const char* directory, const char* fileNameExtension)
+{
+    struct dirent* entry = nullptr;
+    DIR* directoryStream = nullptr;
+    int extensionLength = nullptr != fileNameExtension ? strlen(fileNameExtension) : 0;
+    char* fileName;
+    int fileNameLength;
+    char* lastCharacters;
+    std::vector<std::string> result;
+
+    directoryStream = opendir(directory);
+    if (nullptr != directoryStream)
+    {
+        while ((entry = readdir(directoryStream)))
+        {
+            fileName = entry->d_name;
+            if ((0 == strcmp(fileName, ".")) || (0 == strcmp(fileName, "..")))
+            {
+                continue;
+            }
+
+            if (nullptr == fileNameExtension)
+            {
+                result.push_back(fileName);
+            }
+            else
+            {
+                fileNameLength = strlen(fileName);
+                if (fileNameLength >= extensionLength)
+                {
+                    lastCharacters = &fileName[fileNameLength-extensionLength];
+                    if (0 == strcmp(lastCharacters, fileNameExtension))
+                    {
+                        result.push_back(fileName);
+                    }
+                }
+            }
+        }
+        closedir(directoryStream);
+    }
+    else if (IsFullLoggingEnabled())
+    {
+        OsConfigLogError(PmcLog::Get(), "Failed to open directory %s, cannot list files", directory);
+    }
+
+    return result;
+}
+
+int PmcBase::ConfigureSources(const std::map<std::string, std::string> sources)
+{
+    int status = MMI_OK;
+
+    for (auto& source : sources)
+    {
+        m_executionState.SetExecutionState(ExecutionState::StateComponent::running, ExecutionState::SubStateComponent::modifyingSources, source.first);
+        std::string sourceFileName = source.first + ".list";
+        std::string sourcesFilePath = m_sourcesConfigurationDirectory + sourceFileName;
+        OsConfigLogInfo(PmcLog::Get(), "Starting to configure source(s) file: %s", sourcesFilePath.c_str());
+
+        // Delete file when provided map value is empty
+        if (source.second.empty())
+        {
+            if (FileExists(sourcesFilePath.c_str()))
+            {
+                status = remove(sourcesFilePath.c_str());
+            }
+            else
+            {
+                OsConfigLogInfo(PmcLog::Get(), "Nothing to delete. Source(s) file: %s does not exist", sourcesFilePath.c_str());
```

Comment:
> OsConfigLogInfo(PmcLog::Get(), "Nothing to delete. Source(s) file: %s does not exist", sourcesFilePath.c_str()); [](http://example.com/codeflow?start=15&length=113)

Don't log this. Or only log if full logging is enabled - but only if this is useful.